static/app/views/automations/components/automationBuilder.tsx

```diff
@@ -11,25 +11,139 @@ import {IconAdd, IconDelete, IconMail} from 'sentry/icons';
+      }
+      return incompatibleConditions;
+    }
+  }
+
+  // Find incompatible conditions for ANY_SHORT_CIRCUIT and ALL logic types
+  for (const condition of actionFilter.conditions) {
+    const isInvalidFrequency =
+      frequencyTypes.includes(condition.type) && condition.comparison.value >= 1;
+    const isInvalidAgeComparison =
+      condition.type === DataConditionType.AGE_COMPARISON &&
+      condition.comparison.comparison_type === AgeComparison.OLDER;
+    const isInvalidIssueOccurence =
+      condition.type === DataConditionType.ISSUE_OCCURRENCES &&
+      condition.comparison.value > 1;
+
+    if (isInvalidFrequency || isInvalidAgeComparison || isInvalidIssueOccurence) {
+      incompatibleConditions.push(condition.id);
+    }
+  }
+
+  // If the logic type is ANY_SHORT_CIRCUIT and any of the conditions are valid, consider the action filter valid
+  if (
+    actionFilter.logicType === DataConditionGroupLogicType.ANY_SHORT_CIRCUIT &&
+    incompatibleConditions.length !== actionFilter.conditions.length
+  ) {
+    return [];
+  }
+
+  return incompatibleConditions;
+}
+
 export default function AutomationBuilder() {
   const {state, actions} = useAutomationBuilderContext();
   const organization = useOrganization();
   const api = useApi();
 
-  // fetch org members for SelectMembers dropdowns
+  // Fetch org members for SelectMembers dropdowns
   useEffect(() => {
     fetchOrgMembers(api, organization.slug);
   }, [api, organization]);
 
+  const {conflictingTriggers, conflictingActionFilters} = useMemo((): {
+    conflictingActionFilters: Record<string, string[]>;
+    conflictingTriggers: string[];
+  } => {
+    // First check for conflicting trigger conditions
+    if (state.triggers.logicType === 'all' && state.triggers.conditions.length > 1) {
+      return {
+        conflictingTriggers: state.triggers.conditions.map(condition => condition.id),
+        conflictingActionFilters: {},
+      };
+    }
+
+    // Check for first seen event condition
+    const firstSeenId = state.triggers.conditions.find(
+      condition => condition.type === DataConditionType.FIRST_SEEN_EVENT
+    )?.id;
+
+    const conflictingConditions: Record<string, string[]> = {};
+    let hasConflictingActionFilters = false;
+
+    // First seen event condition does not cause conflicts if the logic type is ANY_SHORT_CIRCUIT and there are multiple trigger conditions
+    if (
+      firstSeenId &&
+      !(
+        state.triggers.logicType === DataConditionGroupLogicType.ANY_SHORT_CIRCUIT &&
+        state.triggers.conditions.length > 1
+      )
+    ) {
+      // Create a mapping of conflicting conditions for each action filter
+      for (const actionFilter of state.actionFilters) {
+        const conflicts = findConflictingActionFilterConditions(actionFilter);
+        conflictingConditions[actionFilter.id] = conflicts;
+        if (conflicts.length > 0) {
+          hasConflictingActionFilters = true;
```

Comment:
feels like we could return here

Reply:
it makes sense to me to highlight all conflicting action filter blocks, but we can talk to design

![image](https://github.com/user-attachments/assets/45648a8a-21fb-4398-90b6-049114005f2e)